logserver/src/main/java/com/yahoo/logserver/handlers/archive/FilesArchived.java

```diff
@@ -45,6 +45,8 @@ private void waitForTrigger(long milliS) throws InterruptedException {
 
     private void run() {
         try {
+            // Sleep some time before first maintenance, unit test depend on files not being removed immediately
+            Thread.sleep(1000);
```

Comment:
Sounds like imperfect unit tests.
Is the delay a little short ?

Reply:
Running the test failed every time for me until I added the delay and worked every time afterwards. On a fast machine I guess it usually works.